src/datumaro/plugins/configurable_validator.py

```diff
@@ -0,0 +1,967 @@
+        )
+        return parser
+
+    def __init__(
+        self,
+        tasks: List[TaskType] = [
+            TaskType.classification,
+            TaskType.detection,
+            TaskType.segmentation,
+        ],
+        warnings: Set[DatasetValidationError] = ALL_WARNINGS,
+        few_samples_thr=None,
+        imbalance_ratio_thr=None,
+        far_from_mean_thr=None,
+        dominance_ratio_thr=None,
+        topk_bins=None,
+    ):
+        self.tasks = tasks
+        self.warnings = warnings
+
+        self.init_flag = False
+        self.all_stats = {task: None for task in tasks}
+
+        self.few_samples_thr = few_samples_thr if few_samples_thr else self.DEFAULT_FEW_SAMPLES_THR
+        self.imbalance_ratio_thr = (
+            imbalance_ratio_thr if imbalance_ratio_thr else self.DEFAULT_IMBALANCE_RATIO_THR
+        )
+        self.far_from_mean_thr = (
+            far_from_mean_thr if far_from_mean_thr else self.DEFAULT_FAR_FROM_MEAN_THR
+        )
+        self.dominance_thr = (
+            dominance_ratio_thr if dominance_ratio_thr else self.DEFAULT_DOMINANCE_RATIO_THR
+        )
+        self.topk_bins_ratio = topk_bins if topk_bins else self.DEFAULT_TOPK_BINS
+
+    def _init_stats_collector(self, label_categories):
+        for task in self.tasks:
+            kwargs = {
+                "few_samples_thr": self.few_samples_thr,
+                "imbalance_ratio_thr": self.imbalance_ratio_thr,
+                "far_from_mean_thr": self.far_from_mean_thr,
+                "dominance_thr": self.dominance_thr,
+                "topk_bins_ratio": self.topk_bins_ratio,
+            }
+            if task == TaskType.classification:
+                self.all_stats[task] = ClsStats(
+                    label_categories=label_categories, warnings=self.warnings, **kwargs
+                )
+            elif task == TaskType.detection:
+                self.all_stats[task] = DetStats(
+                    label_categories=label_categories, warnings=self.warnings, **kwargs
+                )
+            elif task == TaskType.segmentation:
+                self.all_stats[task] = SegStats(
+                    label_categories=label_categories, warnings=self.warnings, **kwargs
+                )
+        self.init_flag = True
+
+    def _get_stats_collector(self, ann_type):
+        if TaskType.classification in self.tasks and ann_type == AnnotationType.label:
+            return self.all_stats[TaskType.classification]
+        elif TaskType.detection in self.tasks and ann_type == AnnotationType.bbox:
+            return self.all_stats[TaskType.detection]
+        elif TaskType.segmentation in self.tasks and ann_type in [
+            AnnotationType.mask,
+            AnnotationType.polygon,
+            AnnotationType.ellipse,
+        ]:
+            return self.all_stats[TaskType.segmentation]
+        else:
+            return None
+
+    def compute_statistics(self, dataset):
+        if not self.init_flag:
```

Comment:
Do we need this flag? I guess that this class has configurations when it is created. With following these configuration and taking the `dataset` in this function argument, it produce statistics according to those two inputs. Therefore, we should initialize statistics carriers (self.all_stats) every time we take `dataset`, right?

Reply:
you meant that we have to compute statistics from scratch when dataset is passed everytime, is this correct?

Reply:
Yes. However, if you imagine the functionality to update incrementally `all_stats` by calling `compute_statistics()` again and again for the sequence of `dataset`, I think it is better to introduce `all_stats` as the function argument (not member variables) for the more understandable data flow in this function.

Reply:
I have removed init_flag following your opinion.